modules/apps/questions/questions-web/src/main/java/com/liferay/questions/web/internal/upgrade/v1_0_1/AddNewServicesSignaturesInSAPEntryUpgradeProcess.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.questions.web.internal.upgrade.v1_0_1;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.security.service.access.policy.service.SAPEntryService;
+
+/**
+ * @author Carlos Correa
+ */
+public class AddNewServicesSignaturesInSAPEntryUpgradeProcess
+	extends UpgradeProcess {
+
+	public AddNewServicesSignaturesInSAPEntryUpgradeProcess(
+		SAPEntryService sapEntryService) {
+
+		_sapEntryService = sapEntryService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		String headlessDeliveryPackage =
+			"com.liferay.headless.delivery.internal.resource.v1_0.";
+
+		String allowedServiceSignatures = StringBundler.concat(
```

Comment:
Hey @ccorreagg, can users modify the content of this SAP? If so I think we might be overwriting in completely. Would it be safer to just append the new values to whatever is already on the DB?

Reply:
The user can modify whatever in any Service Access Policy as there is a menu to do it.

However, the service access policy is an automatic SAP that:

- It is **inserted** when the "Enable Anonymous Read" check is enabled and the Questions configuration is saved.
- It is **deleted** when the "Enable Anonymous Read" check is disabled and the Questions configuration is saved.

For that, it seems the service access policy is being managed by Liferay processes entirely.

What do you think?

![image](https://user-images.githubusercontent.com/32699538/236137644-bfa3f601-f334-4233-b7a7-24a96a5313aa.png)

Reply:
The SAP contains a scoped list of JAX-RS resources that Questions accesses to. I think the lifecycle of the SAP seems to be entirely managed by Questions (by creating and deleting it). For example, if the check is disabled, the whole SAP is deleted instead of just removing the list of resources; for that, I think we are following a similar behaviour.

Reply:
Ok go forward.